ui-speedspacechart/src/components/SpeedSpaceChart.tsx

```diff
@@ -1,36 +1,89 @@
-import React, { useState } from 'react';
-
 import CurveLayer from './layers/CurveLayer';
 import FrontInteractivityLayer from './layers/FrontInteractivityLayer';
 import { type ConsolidatedPositionSpeedTime, OsrdSimulationState } from '../types/simulationTypes';
+import React, { useState } from 'react';
 import type { Store } from '../types/chartTypes';
 import AxisLayerX from './layers/AxisLayerX';
 import AxisLayerY from './layers/AxisLayerY';
+import TickLayerX from './layers/TickLayerX';
+import TickLayerY from './layers/TickLayerY';
+import MajorGridY from './layers/MajorGridY';
+import StepLayer from './layers/StepLayer';
+import ReticleLayer from './layers/ReticleLayer';
+import { resetZoom } from './helpers/layersManager';
 
 export type SpeedSpaceChartProps = {
   width: number;
   height: number;
+  backgroundColor: string;
   data: OsrdSimulationState;
 };
 
-export const SpeedSpaceChart = ({ width, height, data }: SpeedSpaceChartProps) => {
+const SpeedSpaceChart = ({ width, height, backgroundColor, data }: SpeedSpaceChartProps) => {
   const [store, setStore] = useState<Store>({
     speed: data.consolidatedSimulation[0].speed as ConsolidatedPositionSpeedTime[],
-    ratio: 1,
+    stops: data.simulation.present.trains[0].base.stops,
+    electrification: data.simulation.present.trains[0].electrification_ranges,
+    slopes: data.simulation.present.trains[0].slopes,
+    ratioX: 1,
     leftOffset: 0,
+    cursor: {
+      x: null,
+      y: null,
+    },
   });
 
+  const [showDetailsBox, setShowDetailsBox] = useState(false);
+
+  const reset = () => {
+    setStore((prev) => ({
+      ...prev,
+      ratioX: 1,
+      leftOffset: 0,
+    }));
+    resetZoom();
+  };
+
   return (
-    <div className="bg-white" style={{ width: `${width}px`, height: `${height}px` }} tabIndex={0}>
-      <CurveLayer width={width - 60} height={height - 35} store={store} />
+    <div
+      style={{
+        width: `${width}px`,
+        height: `${height}px`,
+        backgroundColor: `${backgroundColor}`,
+      }}
+      tabIndex={0}
+    >
+      <div className="flex justify-end absolute mt-8 ml-2" style={{ width: width }}>
+        <button
+          className="bg-blue-600 hover:bg-blue-700 text-white-100 p-1 mr-6 z-10 rounded-full w-8 h-8"
+          onClick={() => reset()}
+        >
+          &#8617;
+        </button>
+      </div>
+      <CurveLayer width={width - 60} height={height - 80} store={store} />
```

Comment:
Maybe create a new variable or a comment that explains the subtraction of width and height. ( CurveLayer, StepLayer, FrontInteractivityLayer)

Reply:
done